src/world/common/UnkNpcAIFunc13_2.inc.c

```diff
@@ -14,13 +14,13 @@ void N(UnkNpcAIFunc13_2)(ScriptInstance* script, NpcAISettings* aiSettings, s32
     tempAngle = atan2(npc->pos.x, npc->pos.z, gPlayerStatusPtr->position.x, gPlayerStatusPtr->position.z);
     angleDiff = get_clamped_angle_diff(npc->yaw, tempAngle);
 
-    if (aiSettings->unk_10 < fabsf(angleDiff)) {
+    if (aiSettings->unk_10.s < fabsf(angleDiff)) {
```

Comment:
This has to be loaded as if it were an s32?? 🤢
I'm fairly certain it should always be f32, not X32... maybe try some casts

Reply:
The problem is in this function:

https://github.com/pmret/papermario/blob/b84a86d8eb95ca9cb4a85a5b31fa631e825fa897/src/world/area_arn/arn_02/events.c#L269-L286

For `func_800490B4`, unk_10 is loaded and saved with `lw`, meaning that to be a float I have to change the arguments for the function to also be float. Doing that breaks pretty much every single include. I got through fixing a bunch of them but got bored, and went this "simpler" way instead. Look into `func_800490B4` if you want and let me know if it should be a flaot arg or not.

Reply:
`func_800490B4`'s assembly has this line in the prologue:
```s
/* 244C4 800490C4 4487C000 */  mtc1      $a3, $f24
```
So yes, that arg needs to be float... so we shouldn't be doing it the 'simpler' way

Reply:
Ok, I forgot why this didn't work, but now trying to fix it again I see.

So unk_10 is currently typed as `s32` in the repo. It's used here in that function so you say it must be float, however in the linked code you gave from `UnkNpcAIFunc13_2.inc.c`, it's required to load an s32, convert it to float and then do the checking. Here's the diff of that `if (aiSettings->unk_10 < fabsf(angleDiff)) {` line:

![Untitled-1](https://user-images.githubusercontent.com/34639600/112767658-ee308b00-900f-11eb-85bc-97ec9a94bc7f.png)

If we keep it as s32 then there's an issue with the `func_800490B4` call again, as s32 forces a cvt.

Reply:
Yeah, you seem to be right there... X32 feels like the wrong abstraction to me but it's fine until we figure out (read: decomp) all these functions that use `NPCAISettings`.

Reply:
Feel free to leave this as-is in the PR

Reply:
(it's also possible this is purely an `s32` and was typed wrongly as a float? iirc all we've seen so far is `0.0f`, which is the same as `0`)

Reply:
Not sure how that could be, because I was showing that we need both s32 and f32 behaviour from it. If it was typed as  float we should see lwc1 everywhere and the screenshot above should be matching.